src/Documents/Session/IDocumentQuery.ts

```diff
@@ -1,50 +1,48 @@
 import {QueryOperator} from "./QueryOperator";
 import {IRavenObject} from "../../Database/IRavenObject";
 import {DocumentConstructor, DocumentType} from "../Conventions/DocumentConventions";
 
 export interface IDocumentQueryOptions<T> {
-  documentType?: DocumentType<T>, 
+  documentType?: DocumentType<T>,
   indexName?: string;
   usingDefaultOperator?: QueryOperator;
-  waitForNonStaleResults?: boolean;
+  WaitForNonStaleResults?: boolean;
   includes?: string[];
   nestedObjectTypes?: IRavenObject<DocumentConstructor>;
   withStatistics?: boolean;
+  fromCollection?: boolean;
+  queryParameters?: object;
 }
 
 export interface IDocumentQuery<T> {
-  or: IDocumentQuery<T>;
-  and: IDocumentQuery<T>;
-  not: IDocumentQuery<T>;
+  first(callback?: EntityCallback<T>): Promise<T>;
+  count(callback?: EntitiesCountCallback): Promise<number>;
+  take(docsCount: number);
+  skip(skipCount: number);
+  andAlso();
+  orElse();
+  negateNext();
   selectFields(...args: string[]): IDocumentQuery<T>;
-  search(fieldName: string, searchTerms: string | string[], escapeQueryOptions?: EscapeQueryOption, boost?: number): IDocumentQuery<T>;
-  where(conditions: IDocumentQueryConditions): IDocumentQuery<T>;
-  whereEquals<V extends LuceneValue>(fieldName: string, value: V, escapeQueryOptions?: EscapeQueryOption): IDocumentQuery<T>;
-  whereEndsWith(fieldName: string, value: string): IDocumentQuery<T>;
-  whereStartsWith(fieldName: string, value: string): IDocumentQuery<T>;
-  whereIn<V extends LuceneValue>(fieldName: string, values: V[]): IDocumentQuery<T>;
-  whereBetween<V extends LuceneValue>(fieldName: string, start?: V, end?: V): IDocumentQuery<T>;
-  whereBetweenOrEqual<V extends LuceneValue>(fieldName: string, start?: V, end?: V): IDocumentQuery<T>;
-  whereGreaterThan<V extends LuceneValue>(fieldName: string, value: V): IDocumentQuery<T>;
-  whereGreaterThanOrEqual<V extends LuceneValue>(fieldName: string, value: V): IDocumentQuery<T>;
-  whereLessThan<V extends LuceneValue>(fieldName: string, value: V): IDocumentQuery<T>;
-  whereLessThanOrEqual<V extends LuceneValue>(fieldName: string, value: V): IDocumentQuery<T>;
-  whereIsNull(fieldName: string): IDocumentQuery<T>;
-  whereNotNull(fieldName: string): IDocumentQuery<T>;
-  orderBy(fieldsNames: string | string[]): IDocumentQuery<T>;
-  orderByDescending(fieldsNames: string | string[]): IDocumentQuery<T>;
-  andAlso(): IDocumentQuery<T>;
-  orElse(): IDocumentQuery<T>;
-  negateNext(): IDocumentQuery<T>;
-  take(docsCount: number): IDocumentQuery<T>;
-  skip(skipCount: number): IDocumentQuery<T>;
+  search(from: string, searchTerms: string | string[], boostField, boostValue, boostExpression, count): IDocumentQuery<T>;
+  where(conditions: IDocumentQueryConditions);
+  whereEquals<V extends RQLValue>(field: string, value: V): IDocumentQuery<T>;
+  endsWith<V extends RQLValue>(field: string, value: V): IDocumentQuery<T>;
+  exact<V extends RQLValue>(field: string, value: V): IDocumentQuery<T>;
```

Comment:
Exact should not be a method. It's just an option of whereEquals. Please see https://github.com/ravendb/ravendb/blob/v4.0/test/SlowTests/MailingList/Troy.cs#L50